zeebe/broker/src/main/java/io/camunda/zeebe/broker/transport/commandapi/CommandApiRequestHandler.java

```diff
@@ -47,6 +49,18 @@ protected ActorFuture<Either<ErrorResponseWriter, CommandApiResponseWriter>> han
         handle(partitionId, requestId, requestReader, responseWriter, errorWriter));
   }
 
+  public void onRecovered(final ReadonlyStreamProcessorContext context) {
+    processingPaused = false;
```

Comment:
:x: Accessing `processingPaused` must be thread-safe. This is because the the listeners `onPaused, onResumed` etc. are being called by the StreamProcessor. But reading this state is done in the `CommandApi`. They both are different actors meaning they can be running on different threads. So directly updating the field can cause race-conditions. The way to solve this is wrapping the update to this field in `actor::submit`. This might be confusing at the beginning. But we can discuss this f2f.